src/services/cube-handler.ts

```diff
@@ -1303,32 +1304,12 @@ export const createBaseCube = async (datasetId: string, endRevisionId: string):
     logger.debug(rawViewSQL);
     await quack.exec(rawViewSQL);
   }
-  const tmpFile = tmp.tmpNameSync({ postfix: '.db' });
-  try {
-    logger.debug(`Writing memory database to disk at ${tmpFile}`);
-    await quack.exec(`ATTACH '${tmpFile}' as outDB (BLOCK_SIZE 16384);`);
-    await quack.exec(`COPY FROM DATABASE memory TO outDB;`);
-    await quack.exec('DETACH outDB;');
-  } catch (err) {
-    logger.error(err, `Failed to write memory database to disk with error: ${err}`);
-    const error = new CubeValidationException('Failed to write memory database to disk');
-    error.type = CubeValidationType.CubeCreationFailed;
-    throw error;
-  } finally {
-    const end = performance.now();
-    const functionTime = Math.round(end - functionStart);
-    const buildTime = Math.round(end - buildStart);
-    logger.warn(`Cube function took ${functionTime}ms to complete and it took ${buildTime}ms to build the cube.`);
-    await quack.close();
-  }
-  // Pass the file handle to the calling method
-  // If used for preview you just want the file
-  // If it's the end of the publishing step you'll
-  // want to upload the file to the data lake.
   const end = performance.now();
   const functionTime = Math.round(end - functionStart);
   const buildTime = Math.round(end - buildStart);
   logger.warn(`Cube function took ${functionTime}ms to complete and it took ${buildTime}ms to build the cube.`);
+  await quack.close();
+  await new Promise((f) => setTimeout(f, DUCKDB_WRITE_TIMEOUT));
```

Comment:
can we make this a function? e.g. 

```
await safelyCloseDuckDb(quack);
```

```
const safelyCloseDuckDb = async (quack) => {
  await quack.close();
  return new Promise((f) => setTimeout(f, DUCKDB_WRITE_TIMEOUT));
};
```

Reply:
Switched to using a function that combines to the two calls.